src/main/java/com/fasterxml/jackson/databind/ser/std/NumberSerializer.java

```diff
@@ -55,6 +73,21 @@ public JsonSerializer<?> createContextual(SerializerProvider prov,
         if (format != null) {
             switch (format.getShape()) {
             case STRING:
+                if (format.hasPattern()) {
+                    DecimalFormat decimalFormat;
+                    try {
+                        if (format.hasLocale()) {
+                            decimalFormat = new DecimalFormat(format.getPattern(),
+                                    DecimalFormatSymbols.getInstance(format.getLocale()));
+                        } else {
+                            decimalFormat = new DecimalFormat(format.getPattern());
+                        }
+                    } catch (IllegalArgumentException e) {
+                        return prov.reportBadDefinition(handledType(),
+                                String.format("Invalid `DecimalFormat`: \"%s\"", format.getPattern()));
```

Comment:
```suggestion
                                String.format("Invalid 'DecimalFormat': \"%s\"", format.getPattern()));
```

Can we use single quote instead of backtick? I looked around the codebase and in most cases (not all) single quote is used.